src/mscorlib/src/System/DateTime.cs

```diff
@@ -461,7 +461,7 @@ public DateTime(int year, int month, int day, Calendar calendar)
         // parts of the result are the same as those of this DateTime.
         //
         public DateTime AddYears(int value) {
-            if (value < -10000 || value > 10000) throw new ArgumentOutOfRangeException("years", Environment.GetResourceString("ArgumentOutOfRange_DateTimeBadYears"));
+            if (value < -10000 || value > 10000) throw new ArgumentOutOfRangeException(nameof(value), Environment.GetResourceString("ArgumentOutOfRange_DateTimeBadYears"));
```

Comment:
@stephentoub one possible fix that comes to mind will be to change the name of the argument here to `years` and still use the `nameof(years)` so that way the tests pass and the `nameof` enhancement still remains. What do you think?